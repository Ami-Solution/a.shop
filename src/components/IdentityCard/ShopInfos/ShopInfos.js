import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import tokens from '../../../styles/tokens';
import { H1 } from '../../Headings';

const BottomWrapper = styled.div`
  padding: ${tokens.spaces.s};
  background: ${tokens.colors.grey.lightest};
`;

const Adress = styled.p`
  color: ${tokens.colors.blue};
  margin-top: ${tokens.spaces.xxs};
  font-weight: 300;
`;

const Desc = styled.p`
  margin-top: ${tokens.spaces.xxs};
  font-size: ${tokens.fontSizes.xs};
  margin-bottom: 0;
  font-weight: 300;
`;

const ShopInfos = ({ title, adress, description }) => (
  <BottomWrapper>
    <H1>{title}</H1>
    <Adress>{adress}</Adress>
    <Desc>{description}</Desc>
  </BottomWrapper>
);

ShopInfos.propTypes = {
  title: PropTypes.string.isRequired,
  adress: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
};

export default ShopInfos;
