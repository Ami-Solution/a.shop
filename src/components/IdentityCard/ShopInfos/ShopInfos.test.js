/* global describe it expect */
import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ShopInfos from './index';

configure({ adapter: new Adapter() });

describe('Component IdentityCard/ShopInfos', () => {
  const component = renderer
    .create(<ShopInfos
      title="Welcome"
      adress="31 rue de Cotte 75012 Paris"
      description="Vente d'articles cuisine et dÃ©co pour toute la maison."
    />);

  const tree = component.toJSON();

  it('should be render without crash', () => {
    expect(tree).toMatchSnapshot();
  });
});
