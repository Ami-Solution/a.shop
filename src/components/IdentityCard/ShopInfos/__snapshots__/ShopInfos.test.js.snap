// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component IdentityCard/ShopInfos should be render without crash 1`] = `
<div
  className="sc-gzVnrw kjdDyw"
>
  <h1
    className="sc-htpNat iRBdXH"
  >
    Welcome
  </h1>
  <p
    className="sc-htoDjs jQUsuP"
  >
    31 rue de Cotte 75012 Paris
  </p>
  <p
    className="sc-dnqmqq eAQBFl"
  >
    Vente d'articles cuisine et d√©co pour toute la maison.
  </p>
</div>
`;
